File: py/py_tests/data_structure_tests/deque_test.py
import unittest
from py.data_structures.Deque import Deque

class DequeTests(unittest.TestCase):
    def test1(self):
        obj1 = Deque()
        self.assertRaises(IndexError, obj1.pop_first)
        self.assertRaises(IndexError, obj1.pop_last)
        self.assertRaises(IndexError, obj1.peek_first)
        self.assertRaises(IndexError, obj1.peek_last)

    def test2(self):
        obj2 = Deque()
        for i in range(150):
            obj2.add_first(i)
            self.assertEqual(obj2.peek_first(), i)
            self.assertEqual(obj2.peek_last(), 0)
        

    

if __name__ == "__main__":
    unittest.main()

File: js/tests/data_structure_tests/deque.test.js
import {Deque} from "../../data structures/Deque.js";

test('test 1', () => {
    const obj1 = new Deque();
    expect(obj1.popFirst).toThrow();
    expect(obj1.popLast).toThrow();
    expect(obj1.peekFirst).toThrow();
    expect(obj1.peekLast).toThrow();
})